src/app/applications/applications.controller.app.js

```diff
@@ -47,9 +65,43 @@
     }
 
     function uploadIcon(appName) {
-      rClient.uploadIcon(appName).then(function() {
-          $window.location = '#/app/' + appName + '/overview';
+      $scope.cropImage(function() {
+          var dataUrl = $scope.cropImage.dataUrl;
+          var croppedBlob = Cropper.decode(dataUrl);
+          var croppedImage = new File([croppedBlob], $scope.filename);
+          var formObj = new FormData();
+
+          formObj.append('file', croppedImage);
+          rClient.uploadIcon(appName, formObj).then(function() {
+            $window.location.reload(true);
+          });
+      });
+    }
+
+    function selectFile() {
+      $('input[id=file-id]').click();
+    }
+
+    function onFile() {
+      var input = $('#file-id')[0];
+      var blob = input.files[0];
+      Cropper.encode((file = blob)).then(function(dataUrl) {
```

Comment:
Set a variable (?) while invoking a function is not a good practice IMO.
Why not just `Cropper.encode(blob)`?